instance/management/commands/activity_csv.py

```diff
@@ -123,9 +125,13 @@ def handle(self, *args, **options):  # pylint: disable=too-many-locals
 
                 instance_age = datetime.now(instance.created.tzinfo) - instance.created
 
-                emails = [user.email for user in instance.lms_users.all()]
+                try:
+                    email = instance.betatestapplication_set.get().user.email
+                except BetaTestApplication.DoesNotExist:
+                    email = 'N/A'
+
                 csv_writer.writerow([
-                    public_ip, emails,
+                    public_ip, instance.internal_lms_domain, instance.ref.name, email,
```

Comment:
And to answer the first question - I don't think we have anything to check up on broken instances, so this modification might help serve that purpose.